app/eth2wrap/synthproposer.go

```diff
@@ -204,6 +204,34 @@ func (h *synthWrapper) syntheticProposal(ctx context.Context, slot eth2p0.Slot,
 	return proposal, nil
 }
 
+// fieldExists checks if the given field exists as part of the given error.
+func fieldExists(err error, field zap.Field) bool {
```

Comment:
I wonder if instead of declaring this function here, we should create a `z.Fields()` function that returns `[]z.Field`.

The casting logic seems to be a 1:1 copy off `z.Err`.